pkg/queries/user.go

```diff
@@ -145,6 +145,29 @@ func Register(db *gorm.DB, u *models.User, invitation_code string) error {
 		return errors.Wrap(err, errors.DatabaseError)
 	}
 
+
+
+	
+	//若该用户通过邀请码注册
+	if u.InvitationCode != ""{
```

Comment:
`u.InvitationCode` is the invitation code generated in the current function for the registered user, not someone else's invitation code that the new user used to register. Use the `invitation_code` parameter of this function.